src/slicing.jl

```diff
@@ -0,0 +1,97 @@
+export Slicing, Slicer, SlicedEinsum
+
+struct Slicer
+    log2_sizes::Vector{Float64}   # the size dict after slicing
+    legs::Dict{Int,Float64}   # sliced leg and its original size
+    max_size::Int       # maximum number of sliced legs
+end
```

Comment:
The usage of `Slicer` is confusing for me. 
Do you assume indices are `Int` here? Will it be more general to use
```julia
struct Slicer{T}
    log2_sizes::Dict{T, Float64}   # the size dict after slicing
    legs::Dict{T, Float64}   # sliced leg and its original size
    max_size::Int       # maximum number of sliced legs
end
```
instead?

Reply:
There is no need to support general label type, because before entering in `TreeSA` optimizer, ther is a mapping from general label type to integers (for faster indexing in simulated annealing), this is for integer labels and is only used in this algorithm